### ISSUE TITLE
yoshi@4 incorrectly interprets the https setting

### ISSUE THREAD

**Issue by bt4R9** [wix/yoshi#1180]
# 🐛 Bug Report

### 😯 Current Behavior

It seems like there is a conflict with `https` and `servers.cdn.ssl` settings when it comes to open browser in `projectType=app`.
In my app I start yoshi using this command:
```
yoshi start --server=index-dev.js
```
(which is implicitly has `{ https: false }` setting)
but eventually it automatically opens `https://localhost:3000` which is wrong.
It happens because yoshi rely on this condition
```
const https = cliArgs.https || project.servers.cdn.ssl;
```
which is `true` for me because `project.servers.cdn.ssl` is `true`.

### 🤔 Expected Behavior

I expect that command `yoshi start --server=index-dev.js` should open `localhost` without a secure connection.

### 💁 Possible Solution 

I think we should rely on `cliArgs.https` setting not combination of `cliArgs.https || project.servers.cdn.ssl;` when it comes to open browser.

### 👣 Steps to Reproduce 

```
{
  "scripts": {
    "start": "yoshi start --server=index-dev.js"
  },
  "yoshi": {
    "servers": {
      "cdn": {
        "ssl": true
      }
    },
   }
}
```

### 🌍 Your Environment

```
  System:
    OS: macOS 10.14.1
    CPU: (8) x64 Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
  Binaries:
    Node: 11.2.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.4.1 - /usr/local/bin/npm
    Watchman: Not Found
  Browsers:
    Chrome: Not Found
    Firefox: Not Found
    Safari: 12.0.1
  npmPackages:
    @storybook/core:  4.0.0-alpha.14
    @storybook/react: 4.0.0-alpha.14 => 4.0.0-alpha.14
    babel-preset-yoshi:  4.1.0-rc.4
    eslint-config-yoshi:  4.1.0-rc.4
    eslint-config-yoshi-base:  4.1.0-rc.4
    jest-environment-yoshi-bootstrap:  4.1.0-rc.4
    jest-environment-yoshi-puppeteer:  4.1.0-rc.4
    jest-yoshi-preset: ^4.1.0-rc.4 => 4.1.0-rc.4
    storybook: Not Found
    tslint-config-yoshi:  4.1.0-rc.4
    tslint-config-yoshi-base:  4.1.0-rc.4
    typescript: ^3.1.6 => 3.3.4000
    webpack:  4.29.6
    yoshi-config:  4.1.0-rc.4
    yoshi-helpers:  4.1.0-rc.4
    yoshi-runtime: Not Found

 Yoshi config
{
  "externals": {
    "react": "React",
    "react-dom": "ReactDOM",
    "@sentry/browser": "Sentry"
  },
  "hmr": "auto",
  "servers": {
    "cdn": {
      "ssl": true
    }
  },
  "entry": {
    "billing-and-payments-app": "./client/modules/BillingAndPayments/client.tsx",
    "subscriptions-module": "./client/modules/AccountSubscriptions/components/Module/Module.providers.tsx",
    "account-level-app": "./client/modules/AccountLevel/client.tsx",
    "site-level-app": "./client/modules/SiteSubscriptions/client.tsx",
    "site-level-module": "./client/modules/SiteSubscriptions/components/Module/Module.providers.tsx"
  },
  "projectType": "app"
}
```

**Comment by ranyitz** [comment]
Thanks @bt4R9 That's correct and there is a problem.

I think it can be solved by removing the condition `https ? 'https' : 'http'` and put always `http`.

If the user would like otherwise he/she can use the `--url` options.

https://github.com/wix/yoshi/blob/69e8940575e586feb4c5793fda82bd3ee6481925/packages/yoshi/src/commands/start-app.js#L201

Maybe in the future, we could give a better experience for users that run their servers on `https`.

PRs welcome!

**Comment by bt4R9** [comment]
Hi @ranyitz! Initially I wanted to solve the problem of the wrong scheme in the browser. But later I realised that I just don't want to open the browser at all. So currently in our app we're using `BROWSER=none yoshi start` because in our case automatic browser opening isn't a relevant thing since we have several entry points.
So probably besides that we should introduce a new cli setting that would control this behaviour? Something like `--noOpenBrowser`?

**Comment by ranyitz** [comment]
Just to be clear, There is still a bug here that needs to be solved https://github.com/wix/yoshi/issues/1180#issuecomment-490111978 and it would be awesome if someone would take it.

____________________________________

@bt4R9 Regarding your question, It would be good if we could discuss it in a dedicated issue so it would interfere with this thread. 

I still don't understand why you don't want to open the browser, "multiple entry points" is not enough information for me, can you elaborate?

Environment parameter (`BROWSER=none`) is a good solution for a behavior we don't want to promote. We don't want to promote it because in most cases **we do want the browser during start** (we all develop web applications) and there is probably a different problem in the developer experience that could be solved using a better solution.

**Comment by vozhekhovskyi** [comment]
I and @brumeregan are working on this issue at our guild week

**Comment by ranyitz** [comment]
Fixed on #1303
